### PR TITLE
Making tag required param/ html style edit

### LINKED ISSUE #8: Required Tag Association
#### As a user I want all images to have at least one tag.

__Story__:

The tag feature has been a hit. Our users love being able to filter images by
cats, SR-71s, and "AppFolio tshirts" to name a few. In fact, we have discovered
that images without any tags receive significantly less views than those with
tags. As a result, we have decided to require all images to have at least one
tag.

__Acceptance criteria__:
- [ ] I cannot save an image without adding at least one tag.
- [ ] An error message is associated with the appropriate input field when a
tag is not provided.
- [ ] All images have at least one tag.

__Discussion Topic__:
- [ ] What do you do about existing images that have no tags? 
	- [ ] Pretend this is a production system that has been running for years with thousands of users 
	and zillions of images. You can't just drop your production db and rebuild it. Your solution should consider the following points:
		- [ ] You can't just delete the tagless images.
		- [ ] You shouldn't annoy users	by presenting them with validation errors if they try to open or edit a tagless image.
		- [ ] If your solution modifies your production db, then your solution should be able to un-modify your production db 
			  if you decide to run an older version of your code on the production db. (Hint: what Rails mechanism performs
			  versioned db changes?)

__Dependencies__:
- Image Tags
- Refactor flow tests to use page objects

### PR DESCRIPTION
**Pull request by dhanson11**
Closes #8

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -5,25 +5,69 @@
...
-Image.create(url: 'https://cl.escapehere.com/image/upload/t_tn,f_auto,q_auto,$h_345,$w_840/eh/2017/01/Stockholm-sweden.jpg')
-Image.create(url: 'https://static.independent.co.uk/s3fs-public/thumbnails/image/2019/06/05/13/train-sweden.jpg?w968h681')
-Image.create(url: 'https://upload.wikimedia.org/wikipedia/commons/thumb/8/87/Palace_of_Westminster_from_the_dome_on_Methodist_Central_Hall.jpg/1000px-Palace_of_Westminster_from_the_dome_on_Methodist_Central_Hall.jpg')
-Image.create(url: 'https://static.standard.co.uk/s3fs-public/thumbnails/image/2018/07/10/11/Future-London.jpg')
-Image.create(url: 'https://townsquare.media/site/366/files/2012/12/218.jpg?w=980&q=75')
-Image.create(url: 'https://lonelyplanetwp.imgix.net/2018/12/Duomo-1-f2906b7f5bbb.jpg?crop=entropy&fit=crop&h=421&sharp=10&vib=20&w=748')
-Image.create(url: 'https://everyonehatesmarketers.com/wp-content/uploads/2017/04/Dhh-1.jpg')
+image = Image.find_or_initialize_by(url: 'https://d1yjjnpx0p53s8.cloudfront.net/styles/logo-thumbnail/s3/122012/49ers.jpg?itok=RPNisAtV')
+image.tag_list.add('#rubydev', '#appfolio', '#loldogs')
```
Comment (JingyuZ) [inline db/seeds.rb]: This reminds me...do you also need to update existing data to have tags?

File: app/models/image.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 class Image < ActiveRecord::Base
   acts_as_taggable
   validates :url, presence: true, url: true
+  validates :tag_list, presence: true
```
Comment (JingyuZ) [inline app/models/image.rb]: It would be nice to add a more human readable error message, like `At least one tag is required`.

Comment (dhanson11) [reply app/models/image.rb]: is this something that should be done in the model or should i change the error message in new.html.erb

File: db/seeds.rb (hunk 1 above)
Comment (dhanson11) [reply db/seeds.rb]: i modified seeds but assumed database would be otherwise empty. probably not valid

**Pull request closed without merging**